balancer-js/src/modules/pools/pool-types/concerns/stable/join.concern.ts

```diff
@@ -35,6 +53,36 @@ export class StablePoolJoin implements JoinConcern {
     slippage,
     wrappedNativeAsset,
   }: JoinPoolParameters): JoinPoolAttributes => {
+    this.checkInputs(tokensIn, amountsIn, pool);
+    const sortedValues = this.sortValues({
+      pool,
+      wrappedNativeAsset,
+      tokensIn,
+      amountsIn,
+    });
+
+    const { expectedBPTOut, minBPTOut } = this.calcBptOutGivenExactTokensIn({
+      ...sortedValues,
+      slippage,
+    });
+
+    const encodedData = this.encodeJoinPool({
+      joiner,
+      amountsIn,
+      tokensIn,
+      poolId: pool.id,
+      minBPTOut,
+      ...sortedValues,
+    });
+
+    return {
+      ...encodedData,
+      minBPTOut,
+      expectedBPTOut,
+    };
+  };
+
+  checkInputs = (amountsIn: string[], tokensIn: string[], pool: Pool): void => {
```

Comment:
A little comment here is probably useful for an external dev to document that tokens/amounts must have a value for all pool tokens.